cmdstanpy/utils.py

```diff
@@ -902,6 +912,29 @@ def install_cmdstan(
         if stderr:
             logger.warning(stderr.decode('utf-8').strip())
         return False
+
+    # set path to newly installed version
+    cmdstan_dir = dir
+    if dir is None:
+        cmdstan_dir = os.path.expanduser(os.path.join('~', _DOT_CMDSTAN))
+        if not os.path.exists(cmdstan_dir):
+            cmdstanpy_dir = os.path.expanduser(
+                os.path.join('~', _DOT_CMDSTANPY)
+            )
+            if os.path.exists(cmdstanpy_dir):
+                cmdstan_dir = cmdstanpy_dir
+    if version is None:
+        version = get_latest_cmdstan(cmdstan_dir)
+    else:
+        version = '-'.join(['cmdstan', version])
+    new_cmdstan_path = os.path.join(cmdstan_dir, version)
+
+    # validate, set path directly - avoid circular error messages
+    stanc_bin = os.path.join(new_cmdstan_path, 'bin', 'stanc' + EXTENSION)
+    if not os.path.exists(stanc_bin):
+        logger.warning('CmdStan installation failed - missing stanc compiler')
+        return False
+    os.environ['CMDSTAN'] = new_cmdstan_path
```

Comment:
So, this only works if called from python session.

Next time you open a python, the env var does not exist.

Reply:
right - if you install something somewhere, you'll have to write down where you put it.

Reply:
this fix is to address the reasonable user expectation that if you call a function to install a cmdstan version, then going forward - _in the session_  - that becomes the version of cmdstan that is used.   if you call function `install_cmdstan` without specifying the base dir ($HOME/.cmdstan) or version, you'll get this for free.
if, however, you specify some other base dir, or some version other than the latest, you don't.

so this fix takes into account non-default dir and version, and does the right thing.  trying to keep track of what is done in one session for future sessions is a workflow issue, and that's somebody else's problem.  does this make sense?